### PR TITLE
Verify String::remove in alloc::string

### PR DESCRIPTION
**Pull request by szlee118**
### Intro
This verification harness is designed to validate the behavior of the remove method on Rust's String type. The remove method removes a character at a specified index from a string and returns it. The harness leverages Kani's verification capabilities to ensure correctness under various constraints.

### Key Pointer Functions Related:

- `*const T::add / *mut T::add`:
  - Used in self.vec.as_ptr().add(next) and self.vec.as_mut_ptr().add(idx) to compute the addresses of the source and destination pointers for the copy operation.
- `*const T::sub / *mut T::sub`:
  - Conceptually used for addressing adjustments (e.g., computing len - next).

### Purpose
The primary goal of this harness is to verify:
The remove method correctly decreases the string length by one.
The character removed from the string is valid and remains consistent with the original input constraints.

Resolves #76 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/alloc/src/string.rs [hunk 1]
```diff
@@ -3217,3 +3218,42 @@ impl From<char> for String {
...
+
+        // Convert byte array to a String directly (safe since all are ASCII)
+        let mut s = unsafe { String::from_utf8_unchecked(arr.to_vec()) };
+
+        // Ensure the string is not empty
+        kani::assume(!s.is_empty());
+
+        // Generate a valid index within the bounds of the string
+        let idx: usize = kani::any();
+        kani::assume(idx < s.len());
```
Comment (stogaru) [inline library/alloc/src/string.rs]: Can we use kani::any_where for this:

```rust
let idx: usize = kani::anywhere(|&x| x < s.len())
```

File: library/alloc/src/string.rs [hunk 2]
```diff
@@ -3217,3 +3218,42 @@ impl From<char> for String {
...
+
+        // Call the `remove` function
+        let removed_char = s.remove(idx);
+
+        // Verify the string length decreases correctly
+        assert_eq!(s.len(), arr.len() - 1);
+
+        // Verify the removed character is a valid ASCII character
+        assert!(removed_char.is_ascii());
```
Comment (stogaru) [inline library/alloc/src/string.rs:3266]: Can we add an assert verifying whether the removed character is equal to the character at the index `idx` in the original string?

File: library/alloc/src/string.rs [hunk 3]
```diff
@@ -3217,3 +3218,42 @@ impl From<char> for String {
...
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
+    use core::kani;
+    use crate::string::String;
+
+    #[kani::proof]
+    #[kani::unwind(9)] 
+    fn check_remove() {
+        
+        let arr: [u8; 8] = kani::Arbitrary::any_array();
```
Comment (stogaru) [inline library/alloc/src/string.rs]: 8 is a magic number here. Can we hardcode it and add a comment explaining it?

File: library/alloc/src/string.rs [hunk 4]
```diff
@@ -3217,3 +3218,42 @@ impl From<char> for String {
...
+
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
+    use core::kani;
+    use crate::string::String;
+
+    #[kani::proof]
+    #[kani::unwind(9)]
```
Comment (stogaru) [inline library/alloc/src/string.rs]: I believe this is dependent on the array length. Hardcode it maybe?

Comment (szlee118) [reply library/alloc/src/string.rs]: Unwind expects an integer and cannot be hardcoded.

File: library/alloc/src/string.rs (hunk 3 above)
Comment (szlee118) [reply library/alloc/src/string.rs]: Updated in latest commit.

File: library/alloc/src/string.rs (hunk 1 above)
Comment (szlee118) [reply library/alloc/src/string.rs]: Updated in latest commit.

Comment (stogaru) [reply library/alloc/src/string.rs]: You can remove the kani::assume, you no longer need it as the variable is already restricted using `any_where`

File: library/alloc/src/string.rs (hunk 4 above)
Comment (stogaru) [reply library/alloc/src/string.rs]: OK

File: library/alloc/src/string.rs (hunk 1 above)
Comment (stogaru) [reply library/alloc/src/string.rs]: @szlee118 can you please complete this.

File: library/alloc/src/string.rs (hunk 2 above)
Comment (stogaru) [reply library/alloc/src/string.rs:3266]: @szlee118 can you please complete this. I'll merge it either way by 12 am tonight.

Comment (stogaru) [reply library/alloc/src/string.rs:3266]: You can just store the original char in a variable and compare it with the removed character.

File: library/alloc/src/string.rs (hunk 1 above)
Comment (szlee118) [reply library/alloc/src/string.rs]: updated

File: library/alloc/src/string.rs (hunk 2 above)
Comment (szlee118) [reply library/alloc/src/string.rs:3266]: updated

**Pull request merged**